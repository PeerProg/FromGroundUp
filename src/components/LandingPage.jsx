import React, { useState, useEffect } from 'react';
import requestHandler from '../services/requestHandler'

const LandingPage = () => {
  const [landingPageUrl, setlandingPageUrl] = useState([]);

  useEffect(() => {
    requestHandler.homeUrl().then(res => setlandingPageUrl(res.data.message))
  }, [landingPageUrl])
  return (
    <div className="App">
      <div className="App-intro">
        <h2>
          {landingPageUrl}
        </h2>
      </div>
    </div>
  )
};

export default LandingPage;
