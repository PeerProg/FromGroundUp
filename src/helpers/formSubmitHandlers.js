import swal from 'sweetalert2';
import moment from 'moment';
import { createNewHabit, createNewMilestone } from '../services';

export const createHabit = async ({ name, days }, { setSubmitting }, cb) => {
  try {
    const expiresAt = moment().add(days, 'days').format('MMMM DD YYYY, h:mm:ss a');
    const startsAt = moment().format('MMMM DD YYYY, h:mm:ss a');
    const result = await createNewHabit({ name, startsAt, expiresAt });
    const newHabitObject = {
      name: result.data.name,
      startsAt: result.data.startsAt,
      expiresAt: result.data.expiresAt,
      habitId: result.data.id,
      habitActive: result.data.habitActive,
      createdAt: result.data.createdAt,
      updatedAt: result.data.updatedAt,
      milestones: result.data.milestones
    }
    cb(newHabitObject);
    return result.data.id;
  } catch (err) {
    swal({
      type: 'error',
      position: 'top-end',
      title: err.message,
      toast: true,
      showConfirmButton: false,
      timer: 3000
    });
  }
  setSubmitting(false);
}

export const createMilestone = async (
  { habitId, milestone },
  { setSubmitting },
  cb
) => {
  try {
    const result = await createNewMilestone({ habitId, title: milestone });
    cb({ habitId, data: result.data });
  } catch (err) {
    swal({
      type: 'error',
      position: 'top-end',
      title: err.message,
      toast: true,
      showConfirmButton: false,
      timer: 3000
    });
  }
  setSubmitting(false);
};
