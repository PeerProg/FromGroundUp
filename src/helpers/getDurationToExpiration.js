export default expiryDate => {
  const endDate = new Date(expiryDate);
  const today = new Date();
  const ONE_HOUR_IN_MS = 1000 * 60 * 60;
  if (today > endDate) return 'Expired';
  const RAW_TOTAL_DURATION_IN_HOURS = (endDate - today) / ONE_HOUR_IN_MS;
  const ROUNDED_DURATION_IN_HOURS = Math.round(RAW_TOTAL_DURATION_IN_HOURS);
  const ROUNDED_DURATION_IN_DAYS = Math.round(RAW_TOTAL_DURATION_IN_HOURS / 24);
  return ROUNDED_DURATION_IN_HOURS < 48
    ? `${ROUNDED_DURATION_IN_HOURS} hours`
    : `${ROUNDED_DURATION_IN_DAYS} days`;
};
