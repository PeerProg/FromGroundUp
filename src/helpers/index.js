export { isEmpty } from './helperFunctions';
export { signupValidator, loginValidator } from './validators';
export { invalidEmailObject, validRequestObject, unmatchingPasswordObject } from './testData';
