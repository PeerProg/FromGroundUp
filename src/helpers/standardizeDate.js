import moment from 'moment';

export default (rawDate) => {
  const formattedDate = moment(new Date(rawDate)).format('LLL');
  const elements = formattedDate.split(' ')
  const mutatedTimePortion = elements[3][1] === ':' ? `0${elements[3]}` : elements[3];
  const newElements = elements.map((element, index) => {
    return index === 3 ? mutatedTimePortion : element;
  })
  return newElements.join(' ');
}
