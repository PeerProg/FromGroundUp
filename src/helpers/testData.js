export const invalidEmailObject = {
  username: 'Johnathan',
  email: 'johndoe.com',
  password: 'jonathan',
  confirmPassword: 'jonathan'
};

export const unmatchingPasswordObject = {
  username: 'Johnathan',
  email: 'john@doe.com',
  password: 'jonathan',
  confirmPassword: 'jon'
};

export const validRequestObject = {
  username: 'Johnathan',
  email: 'john@doe.com',
  password: 'jonathan',
  confirmPassword: 'jonathan'
};
