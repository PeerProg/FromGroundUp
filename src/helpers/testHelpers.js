import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from 'react-testing-library';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons';
import {
  faEdit,
  faTrashAlt,
  faPlusCircle,
  faCircle,
  faCheckCircle,
  faTimesCircle,
  faMinusCircle
} from '@fortawesome/free-solid-svg-icons';

library.add(
  fab,
  faEdit,
  faTrashAlt,
  faPlusCircle,
  faCircle,
  faCheckCircle,
  faTimesCircle,
  faMinusCircle
);

export function renderWithRouter(ui) {
  return {
    ...render(<Router>{ui}</Router>)
  };
}
