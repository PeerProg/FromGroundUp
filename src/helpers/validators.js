export const signupValidator = inputObject => {
  let errors = {};
  if (!inputObject.email || !inputObject.username || !inputObject.password) {
    errors.email = !inputObject.email ? 'Email Required' : undefined;
    errors.username = !inputObject.username ? 'Username Required' : undefined;
    errors.password = !inputObject.password ? 'Password Required' : undefined;
  } else if (
    !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(inputObject.email)
  ) {
    errors.email = 'Invalid email address';
  } else if (inputObject.password !== inputObject.confirmPassword) {
    errors.confirmPassword = "Passwords don't match";
  }
  return errors;
};

export const loginValidator = inputObject => {
  let errors = {};
  if (!inputObject.identifier || !inputObject.password) {
    errors.identifier = !inputObject.identifier
      ? 'Username or Email Required'
      : undefined;
    errors.password = !inputObject.password ? 'Password Required' : undefined;
  }
  return errors;
};

export const profilePageValidator = inputObject => {
  let errors = {};
  if (!inputObject.username || !inputObject.email) {
    errors.username = !inputObject.username
      ? 'Username cannot be blank'
      : undefined;
    errors.email = !inputObject.email ? 'Email cannot be blank' : undefined;
  }
  return errors;
};

export const habitNameValidator = inputObject => {
  let errors = {};
  const inputValueName = inputObject.name.trim();
  const inputValueDay = inputObject.days;
  if (!inputValueName || !inputValueDay) {
    errors.name = !inputValueName ? 'Habit name cannot be blank' : undefined;
    errors.days = !inputValueDay ? 'Number of Days cannot be blank' : undefined;
  } else if (inputValueName.length < 3) {
    errors.name = 'Habit name must be 3 characters or more';
  } else if (typeof inputValueDay === 'string') {
    errors.days = 'Number of Days must be a number';
  }
  return errors;
};

export const milestoneValidator = inputObject => {
  let errors = {};
  const inputValueHabit = inputObject.habit.trim();
  const inputValueMilestone = inputObject.milestone;
  if (!inputValueHabit || !inputValueMilestone) {
    errors.habit = !inputValueHabit
      ? 'Habit name field cannot be blank'
      : undefined;
    errors.milestone = !inputValueMilestone
      ? 'Milestone field cannot be blank'
      : undefined;
  } else if (inputValueHabit.length < 3 || inputValueMilestone.length < 3) {
    errors.habit = 'Habit name must be 3 characters or more';
    errors.milestones = 'Milestone must be 3 characters or more';
  }

  return errors;
};
