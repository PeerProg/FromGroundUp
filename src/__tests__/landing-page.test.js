import React from 'react';
import { cleanup } from 'react-testing-library';
import LandingPage from '../components/LandingPage';
import { renderWithRouter } from '../helpers';

afterEach(cleanup);

test('should render <LandingPage /> correctly', () => {
  const { container } = renderWithRouter(<LandingPage />);
  expect(container).toMatchSnapshot();
});
