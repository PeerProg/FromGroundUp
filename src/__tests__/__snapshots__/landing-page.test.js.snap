// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <LandingPage /> correctly 1`] = `
<div>
  <div
    class="card mt-5 mr-auto ml-auto mb-auto text-center"
    style="max-height: 500px;"
  >
    <h1
      class="text-monospace mt-5 mb-8 font-weight-bold"
    >
      Welcome to the habit tracker
    </h1>
    <br />
    <br />
    <div
      class="row pl-5 pr-5 mb-5"
    >
      <div
        class="col card mr-5"
        style="background-color: rgb(241, 241, 241); height: 150px;"
      >
        <span
          class="text-monospace mt-auto mb-auto"
        >
          Create Habits
        </span>
      </div>
      <div
        class="col card mr-5"
        style="background-color: rgb(241, 241, 241); height: 150px;"
      >
        <span
          class="text-monospace mt-auto mb-auto"
        >
          Add milestones
        </span>
      </div>
      <div
        class="col card"
        style="background-color: rgb(241, 241, 241); height: 150px;"
      >
        <span
          class="text-monospace mt-auto mb-auto"
        >
          Track milestones completion
        </span>
      </div>
    </div>
  </div>
</div>
`;
